src/frontend/org/voltdb/utils/MiscUtils.java

```
@@ -88,23 +88,21 @@ public static void copyFile(String fromPath, String toPath) throws Exception {
     /**
      * Serialize a file into bytes. Used to serialize catalog and deployment
      * file for UpdateApplicationCatalog on the client.
+     * Notice that if the file is larger than 2GB, readAllBytes() will throw a OutOfMemoryError.
      *
      * @param path
      * @return a byte array of the file
      * @throws IOException
      *             If there are errors reading the file
      */
     public static byte[] fileToBytes(File path) throws IOException {
-        FileInputStream fin = new FileInputStream(path);
-        byte[] buffer = new byte[(int) fin.getChannel().size()];
+        byte[] bytes = null;
         try {
-            if (fin.read(buffer) == -1) {
-                throw new IOException("File " + path.getAbsolutePath() + " is empty");
-            }
-        } finally {
-            fin.close();
+            bytes = Files.readAllBytes(path.toPath());
+        } catch (IOException ioe) {
+            throw new IOException("File " + path.getAbsolutePath() + " is empty");
```

This error message is no longer accurate. You don't have catch it here, just let it throw.